File: frontend/src/services/userService.js
import api from "../api/api";

export const getUsers = async () => {
    const response = await api.get("/users");
    return response.data;
};

export const createUser = async (user) => {
    const response = await api.post("/users", user);
    return response.data;
};

export const updateUserRole = async (id, role) => {
    const response = await api.put(`/users/${id}/role`, {
        role,
    });

    return response.data;
};

export const deleteUser = async (id) => {
    await api.delete(`/users/${id}`);
};

File: backend/app/services/user_service.py
from sqlalchemy.orm import Session

from app.models.user import User
from app.schemas.user import UserCreate
from app.security.jwt import hash_password


def get_all_users(db: Session):
    return db.query(User).all()


def create_user(db: Session, user: UserCreate):
    # Validate role
    allowed_roles = ["Admin", "Operator", "Viewer"]
    role = user.role if user.role in allowed_roles else "Viewer"

    new_user = User(
        username=user.username,
        email=user.email,
        password=hash_password(user.password),
        role=role,
    )

    db.add(new_user)
    db.commit()
    db.refresh(new_user)

    return new_user


def update_user_role(db: Session, user_id: int, role: str):
    user = db.query(User).filter(User.id == user_id).first()

    if not user:
        return None

    user.role = role

    db.commit()
    db.refresh(user)

    return user


def delete_user(db: Session, user_id: int):
    user = db.query(User).filter(User.id == user_id).first()

    if not user:
        return False

    db.delete(user)
    db.commit()

    return True